### PR TITLE
 adding columns types classes

### PR DESCRIPTION
**Pull request by SNsir**

### CODE REVIEW

File: galite-core/src/main/kotlin/org/kopi/galite/visual/VlibProperties.kt [hunk 1]
```diff
@@ -18,8 +18,70 @@
...
+              e)
+      System.err.println("ERROR: " + e.message)
+      "!$key!"
+    }
+  }
+
+  // ----------------------------------------------------------------------
+  // DATA MEMBERS
+  // ----------------------------------------------------------------------
+  private const val VLIB_PROPERTIES_RESOURCE_FILE = "org/kopi/galite/lib/resource/VlibProperties"
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/visual/VlibProperties.kt:86]: Could you fix this to "resource/org/kopi/galite/VlibProperties"

Comment (SNsir) [reply galite-core/src/main/kotlin/org/kopi/galite/visual/VlibProperties.kt:86]: done

File: galite-core/src/main/kotlin/org/kopi/galite/report/VFixnumColumn.kt [hunk 2]
```diff
@@ -33,41 +33,40 @@ import org.kopi.galite.type.NotNullFixed
...
-                  groups,
-                  function,
-                  VFixnumField.computeWidth(digits, maxScale, null, null),
+        options,
+        align,
+        groups,
+        function,
+        VFixnumField.computeWidth(digits, maxScale, null, null),
         1,
         format ?: VFixedFormat(maxScale, true)) {
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/report/VFixnumColumn.kt]: indentation

File: galite-core/src/main/kotlin/org/kopi/galite/report/VStringColumn.kt [hunk 3]
```diff
@@ -47,15 +47,13 @@ class VStringColumn(ident: String,
...
-   * @param    o1    the first operand of the comparison
-   * @param    o2    the second operand of the comparison
-   * @return    -1 if the first operand is smaller than the second
-   * 1 if the second operand if smaller than the first
-   * 0 if the two operands are equal
+   * @param    object1    the first operand of the comparison
+   * @param    object2    the second operand of the comparison
+   * @return         -1 if the first operand is smaller than the second
+   *           1 if the second operand if smaller than the first
+   *           0 if the two operands are equal
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/report/VStringColumn.kt:54]: fix indentation

File: galite-core/src/main/kotlin/org/kopi/galite/visual/VActor.kt [hunk 4]
```diff
@@ -31,14 +31,17 @@ import org.kopi.galite.l10n.LocalizationManager
...
+                  private val menuSource: String?,
+                  val actorIdent: String,
+                  private val actorSource: String?,
+                  var iconName: String?,
+                  val acceleratorKey: Int,
+                  val acceleratorModifier: Int) : VModel {
+
+  override fun getDisplay(): UActor? {
+    return display
+  }
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/visual/VActor.kt]: to inline function

File: galite-core/src/main/kotlin/org/kopi/galite/visual/VActor.kt [hunk 5]
```diff
@@ -88,9 +91,9 @@ open class VActor(
       menuName == obj.menuName
               && menuItem == obj.menuItem
               && ((iconName == null && obj.iconName == null)
-                   || (iconName != null
-                       && obj.iconName != null
-                       && iconName == obj.iconName))
+              || (iconName != null
+              && obj.iconName != null
+              && iconName == obj.iconName))
     }
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/visual/VActor.kt:97]: fix indentation

File: galite-core/src/main/kotlin/org/kopi/galite/visual/VlibProperties.kt [hunk 6]
```diff
@@ -72,8 +72,8 @@ object VlibProperties {
 
     } catch (e: InconsistencyException) {
       ApplicationContext.reportTrouble("localize Property",
-              "org.kopi.galite.visual.VlibProperties.getString(String key, Object[] params)",
-              e.message!!,
+              "org.kopi.galite.visual.VlibProperties.getString(key: String, params: Any?)",
+              e.message,
               e)
       System.err.println("ERROR: " + e.message)
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/visual/VlibProperties.kt]: remove +

File: galite-core/src/main/kotlin/org/kopi/galite/visual/VlibProperties.kt [hunk 7]
```diff
@@ -83,5 +83,5 @@ object VlibProperties {
   // ----------------------------------------------------------------------
   // DATA MEMBERS
   // ----------------------------------------------------------------------
-  private const val VLIB_PROPERTIES_RESOURCE_FILE = "resource/org/kopi/galite/VlibProperties"
+  private const val VLIB_PROPERTIES_RESOURCE_FILE = "org/kopi/galite/lib/resource/VlibProperties"
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/visual/VlibProperties.kt]: "resource/org/kopi/galite/VlibProperties"

File: galite-core/src/main/kotlin/org/kopi/galite/visual/VlibProperties.kt [hunk 8]
```diff
@@ -72,8 +72,8 @@ object VlibProperties {
 
     } catch (e: InconsistencyException) {
       ApplicationContext.reportTrouble("localize Property",
-              "org.kopi.galite.visual.VlibProperties.getString(String key, Object[] params)",
-              e.message!!,
+              "org.kopi.galite.visual.VlibProperties.getString(key: String, params: Any?)",
+              e.message,
               e)
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/visual/VlibProperties.kt]: check indentation

File: galite-core/src/main/kotlin/org/kopi/galite/report/PExport.kt [hunk 9]
```diff
@@ -190,15 +190,24 @@ abstract class PExport(val table: UTable,
...
+  open fun formatIntegerColumn(column: VReportColumn, index: Int) {}
+
+  internal open fun formatBooleanColumn(column: VReportColumn, index: Int) {}
+
+  internal open fun formatTimeColumn(column: VReportColumn, index: Int) {}
+
+  internal open fun formatTimestampColumn(column: VReportColumn, index: Int) {}
 
   fun getColumnLabel(column: Int): String {
     return model.getAccessibleColumn(table.convertColumnIndexToModel(column))!!.label
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/report/PExport.kt]: to inline function

**Review by h-haddad — APPROVED** [review]

**Pull request merged**